src/emulator/types.ts

```diff
@@ -117,14 +117,18 @@ export class EmulatorLog {
     this.data = this.data || {};
   }
 
-  toString(): string {
-    return JSON.stringify({
-      timestamp: this.timestamp,
-      level: this.level,
-      text: this.text,
-      data: this.data,
-      type: this.type,
-    });
+  toString(pretty = false): string {
```

Comment:
Maybe just two methods: toString() and toPrettyString()

This is a total nit but I find arguments to `toString()` to be pretty strange.